frontend/src/components/editable/EditableText.vue

```diff
@@ -0,0 +1,252 @@
+          <template v-slot:message="{ key, message }">
+            <error-message :message="message" @close="clearMessages"/>
+          </template>
+        </v-text-field>
+      </v-card>
+    </v-menu>
+  </div>
+</template>
+
+<script>
+import ErrorMessage from './ErrorMessage'
+
+export default {
+  name: 'editable-text',
+  components: {
+    ErrorMessage
+  },
+  props: {
+    items: {
+      type: Array
+    },
+    value: {
+      required: true
+    },
+    save: {
+      type: Function,
+      default: () => {}
+    },
+    rules: {
+      type: Array,
+      default: () => []
+    },
+    color: {
+      type: String,
+      default: 'blue-grey darken-2'
+    },
+    readOnly: {
+      type: Boolean,
+      default: false
+    }
+  },
+  data () {
+    return {
+      contentWidth: 500,
+      timeoutId: undefined,
+      contentBounce: false,
+      active: false,
+      loading: false,
+      messages: [],
+      error: undefined,
+      lazyValue: undefined
+    }
+  },
+  computed: {
+    isActive: {
+      get () {
+        return this.active
+      },
+      set (value) {
+        if (this.$refs.content) {
+          const { width } = this.$refs.content.getBoundingClientRect()
+          this.contentWidth = width - 8
+        }
+        this.active = value
+      }
+    },
+    internalValue: {
+      get () {
+        return this.lazyValue
+      },
+      set (value) {
+        this.lazyValue = value
+      }
+    },
+    activatorColor () {
+      if (this.contentBounce) {
+        return 'success'
+      }
+      return this.color
+    },
+    activatorIcon () {
+      if (this.contentBounce) {
+        return 'mdi-check'
+      }
+      if (this.isActive) {
+        return 'mdi-close'
+      }
+      return 'mdi-pencil'
+    }
+  },
+  methods: {
+    clearMessages () {
+      this.messages = []
+    },
+    onCancel () {
+      this.internalValue = this.value
+      this.isActive = false
+    },
+    async onSave (value) {
+      this.loading = this.color
+      try {
+        await this.save(this.internalValue)
+        this.$emit('input', this.internalValue)
+        setImmediate(() => {
+          this.isActive = false
+        })
+        this.contentBounce = true
+        this.timeoutId = setTimeout(() => {
+          this.contentBounce = false
+        }, 1000)
+      } catch (err) {
+        this.messages.push(JSON.stringify([err.message, err.detailedMessage]))
+      } finally {
+        this.loading = false
+      }
+    },
+    focus () {
+      const textField = this.$refs.textField
+      if (textField) {
+        textField.focus()
+      }
+    },
+    reset () {
+      this.messages = []
+      const textField = this.$refs.textField
+      if (textField) {
+        textField.resetValidation()
+      }
+    }
+  },
+  watch: {
+    active (value) {
+      if (value) {
+        clearTimeout(this.timeoutId)
+        this.internalValue = this.value
+        this.$emit('open')
+        setTimeout(() => {
```

Comment:
use setDelayedInputFocus

Reply:
The function `setDelayedInputFocus` behave differently. It also selects the content of an input field. It also does not use the focus method of the textField since this does not select the content. I am not sure if we should select the content in this case.

Reply:
✔️